app/containers/MessageActions.js

```diff
@@ -82,6 +82,10 @@ export default class MessageActions extends React.Component {
 			this.REPLY_INDEX = this.options.length - 1;
 		}
 
+		// Mark as unread
+		this.options.push(I18n.t('Mark_as_unread'));
```

Comment:
Push it to `this.options` only if it's from the same user.

Reply:
how to check it? this.isRoomReadOnly()?

Reply:
You can only mark as unread something you didn't write yourself.
So we need to check if the selected message's author agains current user.
```jsx
// Toggle Auto-translate
props.actionMessage.u && props.actionMessage.u._id !== props.user.id
```